session/txn.go

```diff
@@ -431,9 +437,13 @@ func (s *session) getTxnFuture(ctx context.Context) *txnFuture {
 }
 
 // StmtCommit implements the sessionctx.Context interface.
-func (s *session) StmtCommit() error {
-	defer s.txn.cleanup()
+func (s *session) StmtCommit(memTracker *memory.Tracker) error {
```

Comment:
The statement is finished and the memory is allocated when `StmtCommit`, I think it is meanless to Consume it at https://github.com/pingcap/tidb/blob/d666ed41377d0d9262fe90585a697e375c80b335/session/txn.go#L260 .

Reply:
But stmtBuffer can be GCed after be written to txnBuffer?